writing-visualising/dashboards_rshiny.qmd

```diff
@@ -7,36 +7,148 @@ title: "Creating R Shiny dashboards"
+if(FALSE){
+  # create a pool connection
+  connection <- pool::dbPool(...)
+  
+  # unlazy data collected into memory BEFORE server
+  unlazy_data <- dplyr::tbl(connection, DBI::Id(...)) |>
+    collect()
+  
+  # create a lazy connection to a database
+  lazy_school_table <- dplyr::tbl(connection, DBI::Id(...))
+  
+  # tell shiny to close the connection when the app stops
+  shiny::onStop(function() poolClose(connection))
+  
+  # an example ui to select a school
+  ui <- shiny::fluidPage(
+    shiny::selectInput(
+      inputId = "school"
+    ),
+    shiny::tableOutput("all_schools"),
+    shiny::tableOutput("one_school")
+  )
+  
+  # data collected into memory ON SERVER SIDE
+  server <- function(input, output, session){
+    school_level_data <- reactive({
+      lazy_school_table |>
+        filter(school_name == !!input$school) |>
+        collect()
+    })
+    
+    # table showing all schools
+    output$all_schools <- renderTable({
+      unlazy_data
+    })
+    
+    # table showing filtered school
+    output$one_school <- renderTable({
+      school_level_data()
+    })
+  }
+}
+```
+
+### Levelling up your shiny
+
+This is just the tip of the iceberg. 
+
+[Mastering Shiny](https://mastering-shiny.org/index.html) is a really useful read, especially the [Best Practice](https://mastering-shiny.org/scaling-intro.html) section.
+
+There is a plethora of [awesome shiny extensions](https://github.com/nanxstats/awesome-shiny-extensions) to explore. Keep it accessible, use packages from CRAN/ones you trust and the world is your oyster.
+
+For more **custom user interface**: 
+
+- have a read of [Outstanding User Interfaces with Shiny](https://unleash-shiny.rinterface.com/welcome.html) for how the web stuff works with shiny 
+- for easy theme customisation [Bootstrap Live Customiser](https://www.bootstrap-live-customizer.com/) 
+- For more context of the web bits, [HTML](https://www.w3schools.com/html/default.asp), [CSS](https://www.w3schools.com/css/default.asp) and [Bootstrap](https://www.w3schools.com/bootstrap/bootstrap_ver.asp) give a better understanding of what's under the hood. 
+
+:::{.callout-caution}
+HTML, CSS and Bootstrap are a slippery slope into web development. Remember that if you're working with Shiny then good analysis comes first!
+:::
+
+For **interactive maps**, check out [this section](../learning-development/r.html#example-shiny-app-with-map).
+
+For improving your **tables**, check out [this section](../learning-development/r.html#rendering-tables).
+
+For more advanced shiny knowledge it's worth taking a look at the guide to [engineering production-grade shiny apps](https://engineering-shiny.org/). It's worth reading and considering how much to adopt, remember that just because it sounds clever doesn't make it easy for other analysts to pick up. Read the [Don't sacrifice readability](https://engineering-shiny.org/need-for-optimization.html#dont-sacrifice-readability) section and then work from there.
```

Comment:
I'm a big fan of the "just because it sounds clever, doesn't make it easy for other analysts to pick up" line. I'd be tempted to add a rephrasing of this right near the start of this page in a call-out box (could be any of note, tip, warning or danger...). 

Maybe something along the lines of: "Just because something looks clever, doesn't mean you should do it. Don't over-complicate your user interface or the underlying code. Your priorities should be making something that meets user's needs, whilst also creating something that's manageable for whoever follows you in maintaining your dashboard."